src/persist/disk.rs

```diff
@@ -121,12 +120,13 @@ impl Backend for DiskBackend {
 
             let mut data_ofs = self.data_ofs.lock();
 
```

Comment:
I know it was not introduced by this change, but I find the offset mechanism a bit risky, as writing the file is outside of mutex and the sync between writing and the state of variable data_ofs is not guaranteed IMO. Should we have a wider scope for mutex or read offset from file?